test/integration/helpers.js

```diff
@@ -81,27 +81,30 @@ module.exports = {
     var path;
 
     path = resolveFixturePath(fixturePath);
-    args = args || [];
+    args = (args || []).concat(['--reporter', 'json', path]);
 
     return invokeMocha(
-      args.concat(['--reporter', 'json', path]),
+      args,
       function(err, res) {
-        if (err) return fn(err);
+        if (err) {
+          return fn(err);
+        }
 
+        var result;
         try {
-          var result = toJSONRunResult(res);
-          fn(null, result);
+          result = toJSONRunResult(res);
         } catch (err) {
-          fn(
+          return fn(
             new Error(
               format(
-                'Failed to parse JSON reporter output.\nArgs: %O\nResult:\n\n%O',
-                args,
+                'Failed to parse JSON reporter output: %s\n\nResult:%O',
+                err,
                 res
               )
             )
           );
         }
+        fn(null, result);
```

Comment:
Why move this? Original location was more to the point.
```js
try {
  return fn(null, toJSONRunResult(res));
} catch (err) {
  return fn(
    new Error(
      format(
        'Failed to parse JSON reporter output: %s\n\nResult:%O',
        err,
        res
      )
    )
  );
}
```

Reply:
because the logic was incorrect.  it would catch a potential error thrown out of the `fn` callback.

Reply:
I'll add a comment explaining myself